### ISSUE TITLE
proposal: allow starting the QEMU gdbserver

### ISSUE THREAD

**Issue by lmb** [danobi/vmtest#4]
It would be nice if it was possible to invoke vmtest like so:

```
$ vmtest --debug
```

Which would then proceed to start the [qemu GDB server](https://qemu-project.gitlab.io/qemu/system/gdb.html). (Preferably bound to localhost instead of *.) This makes it really easy to debug kernel issues or bugfixes.

**Comment by danobi** [comment]
Makes sense to me. But rather than over TCP (which would imply bringing up networking), how about orchestrating a serial device that's shared between the host and guest? The gdbserver docs seem to imply that's possible. For kernel gdb server TCP is probably fine (looking at qemu's `-s` flag)

**Comment by lmb** [comment]
I only care about the in-qemu GDB server to be able to debug the kernel, I wasn't aware that it's possible to start one in the guest as well!
